skpro/survival/gradient_boosting_surv/_ngboost_surv.py

```diff
@@ -0,0 +1,339 @@
+        the percent subsample of rows to
+        use in each boosting iteration
+    verbose : boolean, default=True
+        flag indicating whether output
+        should be printed during fitting
+    verbose_eval : int ,default=100
+        increment (in boosting iterations) at
+        which output should be printed
+    tol : float, default = 1e-4
+        numerical tolerance to be used in optimization
+    random_state : int, RandomState instance or None, optional (default=None)
+
+    Returns
+    -------
+        An NGBSurvival object that can be fit.
+    """
+
+    _tags = {
+        "authors": ["ShreeshaM07"],
+        "maintainers": ["ShreeshaM07"],
+        "python_dependencies": "ngboost",
+    }
+
+    def __init__(
+        self,
+        dist="LogNormal",
+        score="LogScore",
+        estimator=None,
+        natural_gradient=True,
+        n_estimators=500,
+        learning_rate=0.01,
+        minibatch_frac=1.0,
+        col_sample=1.0,
+        verbose=True,
+        verbose_eval=100,
+        tol=1e-4,
+        random_state=None,
+    ):
+        self.dist = dist
+        self.score = score
+        self.estimator = estimator
+        self.natural_gradient = natural_gradient
+        self.n_estimators = n_estimators
+        self.learning_rate = learning_rate
+        self.minibatch_frac = minibatch_frac
+        self.col_sample = col_sample
+        self.verbose = verbose
+        self.verbose_eval = verbose_eval
+        self.tol = tol
+        self.random_state = random_state
+
+        super().__init__()
+
+    def _dist_to_ngboost_instance(self, dist):
+        """Convert string to NGBoost object.
+
+        Parameters
+        ----------
+        dist : string
+            the input string for the type of Distribution.
+            It then creates an object of that particular NGBoost Distribution.
+
+        Returns
+        -------
+        NGBoost Distribution object.
+        """
+        from ngboost.distns import LogNormal
+
+        ngboost_dists = {
+            "LogNormal": LogNormal,
+        }
+        # default Normal distribution
+        dist_ngboost = LogNormal
+        if dist in ngboost_dists:
+            dist_ngboost = ngboost_dists[dist]
+
+        return dist_ngboost
+
+    def _fit(self, X, y, C=None):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Changes state to "fitted" = sets is_fitted flag to True
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            feature instances to fit regressor to
+        y : pd.DataFrame, must be same length as X
+            labels to fit regressor to
+        C : pd.DataFrame, optional (default=None)
+            censoring information for survival analysis,
+            same length as X and y
+            should have entries 0 and 1 (float or int)
+            1 = uncensored, 0 = (right) censored
```

Comment:
I think this is switched around.

The `skpro` interface has 1 = censored. So you need to do "one minus" if `C` is not `None`.

Reply:
reason is, we need a unified API, censored always needs to be represented by the same number. Different packages do it differently, but from the `skpro` side we always need to do it the same way.